pkg/aws/core_util.go

```diff
@@ -131,34 +111,87 @@ func (d *Driver) getInstancesFromMachineName(machineName string, providerSpec *a
 			},
 		},
 	}
-
 	runResult, err := svc.DescribeInstances(&input)
+
 	if err != nil {
 		klog.Errorf("AWS plugin is returning error while describe instances request is sent: %s", err)
 		return nil, status.Error(codes.Internal, err.Error())
+	} else {
+		var instances []*ec2.Instance
+		for _, reservation := range runResult.Reservations {
+			instances = append(instances, reservation.Instances...)
+		}
+		return instances, nil
 	}
+}
 
-	for _, reservation := range runResult.Reservations {
-		instances = append(instances, reservation.Instances...)
+// getMatchingInstancesForMachine extracts AWS Instance object for a given machine
+func (d *Driver) getMatchingInstancesForMachine(machine *v1alpha1.Machine, providerSpec *api.AWSProviderSpec, secret *corev1.Secret) ([]*ec2.Instance, error) {
+	var (
+		clusterName string
+		nodeRole    string
+		instances   []*ec2.Instance
+	)
+	svc, err := d.createSVC(secret, providerSpec.Region)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	for key := range providerSpec.Tags {
+		if strings.Contains(key, "kubernetes.io/cluster/") {
+			clusterName = key
+		} else if strings.Contains(key, "kubernetes.io/role/") {
+			nodeRole = key
+		}
+	}
+	instances, err = getMachineInstancesByTagsAndStatus(svc, machine.Name, clusterName, nodeRole)
+	if err != nil {
+		return nil, err
 	}
 	if len(instances) == 0 {
```

Comment:
can you add a log stating that no VM was found using tags/status, now looking for VM using the providerID? This will help in diagnosing cases that we ran into where no VM was found when looking for a VM using tags + status.